### PR TITLE
validator client on rest mode has an inappropriate context deadline for events

### LINKED ISSUE #13763: Validator client: Regular `Event stream interrupted` error
### Describe the bug

BN launch command:
```
bazel run //cmd/beacon-chain --config=release -- --execution-endpoint=http://localhost:8551 --holesky --jwt-secret /Users/manu/OffchainLabs/ethereum/jwt.hex --checkpoint-sync-url https://holesky.beaconstate.info --genesis-beacon-api-url https://holesky.beaconstate.info --accept-terms-of-use --force-clear-db
```

VC launch command:
```
bazel run //cmd/validator -- --holesky --wallet-password-file /Users/manu/Downloads/password.txt --suggested-fee-recipient <xxx> --enable-beacon-rest-api
```

VC logs:

```
[2024-03-18 11:24:02]  INFO client: Submitted new attestations blockRoot=0xe99f7507b82e committeeIndices=[43] pubkeys=[0x84d7fa31925c] slot=1237920 sourceEpoch=38684 sourceRoot=0x7c1100c4ffe3 targetEpoch=38685 targetRoot=0xe99f7507b82e
[2024-03-18 11:24:12] ERROR client: Event stream interrupted error=scanner failed: could not connect: context deadline exceeded (Client.Timeout or context cancellation while reading body)
[2024-03-18 11:24:18]  INFO client: Event stream reconnecting...
[2024-03-18 11:24:18]  INFO client: Starting event stream topics=[head]
[2024-03-18 11:24:18]  INFO Listening to Beacon API events topics=head
[2024-03-18 11:24:40]  INFO client: Submitted new attestations blockRoot=0x0126b80ef44a committeeIndices=[22] pubkeys=[0x93ae299ee8c2] slot=1237923 sourceEpoch=38684 sourceRoot=0x7c1100c4ffe3 targetEpoch=38685 targetRoot=0xe99f7507b82e
[2024-03-18 11:24:48] ERROR client: Event stream interrupted error=scanner failed: could not connect: context deadline exceeded (Client.Timeout or context cancellation while reading body)
[2024-03-18 11:24:54]  INFO client: Event stream reconnecting...
[2024-03-18 11:24:54]  INFO client: Starting event stream topics=[head]
[2024-03-18 11:24:54]  INFO Listening to Beacon API events topics=head
[2024-03-18 11:25:04]  INFO client: Submitted new attestations blockRoot=0x938c1ee5bc0b committeeIndices=[62] pubkeys=[0x9471e0558cb1] slot=1237925 sourceEpoch=38684 sourceRoot=0x7c1100c4ffe3 targetEpoch=38685 targetRoot=0xe99f7507b82e
[2024-03-18 11:25:16]  INFO client: Submitted new attestations blockRoot=0x142631dc4261 committeeIndices=[53] pubkeys=[0x90c409ed5de3] slot=1237926 sourceEpoch=38684 sourceRoot=0x7c1100c4ffe3 targetEpoch=38685 targetRoot=0xe99f7507b82e
[2024-03-18 11:25:24] ERROR client: Event stream interrupted error=scanner failed: could not connect: context deadline exceeded (Client.Timeout or context cancellation while reading body)
[2024-03-18 11:25:28]  INFO client: Submitted new attestations blockRoot=0x52a0d656d18f committeeIndices=[3] pubkeys=[0xb83fc46c5fc9] slot=1237927 sourceEpoch=38684 sourceRoot=0x7c1100c4ffe3 targetEpoch=38685 targetRoot=0xe99f7507b82e
[2024-03-18 11:25:30]  INFO client: Event stream reconnecting...
[2024-03-18 11:25:30]  INFO client: Starting event stream topics=[head]
[2024-03-18 11:25:30]  INFO Listening to Beacon API events topics=head
[2024-03-18 11:25:40]  INFO client: Submitted new attestations blockRoot=0xd69f77bed822 committeeIndices=[36 47 9] pubkeys=[0x8c3683846835 0x80f408a41341 0x8ae1e4efc47f] slot=1237928 sourceEpoch=38684 sourceRoot=0x7c1100c4ffe3 targetEpoch=38685 targetRoot=0xe99f7507b82e
[2024-03-18 11:26:00] ERROR client: Event stream interrupted error=scanner failed: could not connect: context deadline exceeded (Client.Timeout or context cancellation while reading body)
```

Seems to work well without the `--enable-beacon-rest-api` option.

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

 Bug fix

**What does this PR do? Why is it needed?**

Fixes the following error on the validator client while running with the flag `--enable-beacon-rest-api`

```
[2024-03-18 11:26:00] ERROR client: Event stream interrupted error=scanner failed: could not connect: context deadline exceeded (Client.Timeout or context cancellation while reading body)
```

Fixes the following issue to bubble up errors and kill the connection instead of just logging errors continuously 

```
[2024-03-18 11:07:25] ERROR Could not write to response writer error=write tcp 127.0.0.1:3500->127.0.0.1:59590: write: broken pipe
```

**Which issues(s) does this PR fix?**

Fixes #13763

**Other notes for review**

### CODE REVIEW

File: validator/client/beacon-api/beacon_api_validator_client.go [hunk 1]
```diff
@@ -192,7 +193,7 @@ func (c *beaconApiValidatorClient) WaitForChainStart(ctx context.Context, _ *emp
 }
 
 func (c *beaconApiValidatorClient) StartEventStream(ctx context.Context, topics []string, eventsChannel chan<- *event.Event) {
-	eventStream, err := event.NewEventStream(ctx, c.jsonRestHandler.HttpClient(), c.jsonRestHandler.Host(), topics)
+	eventStream, err := event.NewEventStream(ctx, &http.Client{}, c.jsonRestHandler.Host(), topics) // event stream should not be subject to the same settings as other api calls
```
Comment (prestonvanloon) [inline validator/client/beacon-api/beacon_api_validator_client.go:196]: This comment isn't clear, consider making a variable to make it more clear.
```suggestion
	client := &http.Client{} // event stream should not be subject to the same settings as other api calls, so we won't use c.jsonRestHandler.HttpClient()
	eventStream, err := event.NewEventStream(ctx, client, c.jsonRestHandler.Host(), topics) 
```

File: beacon-chain/rpc/eth/events/events.go [hunk 2]
```diff
@@ -497,32 +485,31 @@ func (s *Server) sendPayloadAttributes(ctx context.Context, w http.ResponseWrite
...
+func sendKeepalive(w http.ResponseWriter, flusher http.Flusher) error {
+	return write(w, flusher, ":\n\n")
 }
 
-func write(w http.ResponseWriter, flusher http.Flusher, format string, a ...any) {
+func write(w http.ResponseWriter, flusher http.Flusher, format string, a ...any) error {
 	_, err := fmt.Fprintf(w, format, a...)
 	if err != nil {
-		log.WithError(err).Error("Could not write to response writer")
+		return errors.Wrap(err, "Could not write to response writer")
```
Comment (prestonvanloon) [inline beacon-chain/rpc/eth/events/events.go:511]:
```suggestion
		return errors.Wrap(err, "could not write to response writer")
```
errors shouldn't be capitalized here

File: api/client/event/event_stream.go [hunk 3]
```diff
@@ -106,7 +106,10 @@ func (h *EventStream) Subscribe(eventsChannel chan<- *Event) {
 	var eventType, data string // Variables to store event type and data
 
 	// Iterate over lines of the event stream
-	for scanner.Scan() {
+	for {
+		if ok := scanner.Scan(); !ok {
+			break
+		}
```
Comment (prestonvanloon) [inline api/client/event/event_stream.go:112]: Is this change necessary?

Comment (james-prysm) [reply api/client/event/event_stream.go:112]: i don't think so anymore I think I'll revert this

**Review by prestonvanloon — APPROVED** [review]

**Review by rkapka — DISMISSED** [review]

**Review by rkapka — APPROVED** [review]

**Review by nalepae — APPROVED** [review]

**Pull request merged**